subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/PendingDependencies.java

```diff
@@ -65,4 +65,13 @@ void decreaseHardEdgeCount() {
         assert hardEdges > 0 : "Cannot remove a hard edge when none recorded";
         hardEdges--;
     }
+
+    public boolean shouldReportActivatePending() {
+        for (NodeState affectedComponent : affectedComponents) {
```

Comment:
:o: let's see but I think this can have a significant impact on performance. You could precompute this flag when affected components are registered.

Reply:
Good point, let me change that.